port/cpl_aws.cpp

```diff
@@ -859,6 +859,36 @@ bool VSIS3HandleHelper::GetConfigurationFromEC2(
                                    osPathForOption.c_str(),
                                    "AWS_CONTAINER_CREDENTIALS_RELATIVE_URI", "")
                              : std::string());
+    // coverity[tainted_data]
+    const std::string osECSTokenFile(
+        (osECSFullURI.empty() && osECSRelativeURI.empty())
+            ? std::string()
+            : VSIGetPathSpecificOption(osPathForOption.c_str(),
+                                       "AWS_CONTAINER_AUTHORIZATION_TOKEN_FILE",
+                                       ""));
+
+    // coverity[tainted_data]
+    const std::string osECSTokenValue(
+        (osECSFullURI.empty() && osECSRelativeURI.empty() &&
+         !osECSTokenFile.empty())
+            ? std::string()
+            : VSIGetPathSpecificOption(osPathForOption.c_str(),
+                                       "AWS_CONTAINER_AUTHORIZATION_TOKEN",
+                                       ""));
+
+    std::string osECSToken;
+    if (!osECSTokenFile.empty())
+    {
+        if (!ReadAWSWebIdentityTokenFile(osECSTokenFile, osECSToken))
```

Comment:
`ReadAWSWebIdentityTokenFile` is re-used here to read the container authorization token file. It might be better to either rename `ReadAWSWebIdentityTokenFile` to something more generic, or duplicate its functionality to make this part a bit clearer.

Reply:
Feel free to rename to ReadAWSTokenFile for example